docs/proposals/ldap-group-sync.md

```diff
@@ -0,0 +1,108 @@
+
+OpenShift Resource | LDAP Entry
+------------------ | ----------
+OpenShift Group    | LDAP group
+OpenShift `Group`  | group
+`Group`            |
+___
+
+Linking OpenShift `Groups` to their representations on LDAP servers allows for administrators to have an easier way to manage `Groups` inside OpenShift. This additional functionality will require no additional fields on the OpenShift `Group` object. The sync job will indicate that an OpenShift `Group` mirrors an external representation by setting the `openshift.io/externalRecordProvider` `Annotation` in the `Group` as well as adding a `Label` such as `openshift.io/externalRecordProvider/LDAP` to the `Group`. Other information (e.g. last sync time or resource version) will be stored as `Annotations` on the OpenShift `Group`. Logistical information for the LDAP sync will be stored in an `LDAPSyncOptions` that the administrator will need to provide.
+
+### Use Cases
+1. I want to sync all LDAP groups I've configured in a sync job (e.g. a specific LDAP server configuration) to OpenShift `Groups`
+* I want to sync specific OpenShift `Groups` with their `externalRecordProviders`
+* I want to sync a speficic set of LDAP groups to OpenShift `Groups`
+* I want to run a destructive sync that will remove OpenShift groups that were created or updated by a previous sync if the LDAP server no longer contains their record
+* I want to map an LDAP group to an existing OpenShift `Group`
+
+## Support for LDAP Group Schemas
+The three LDAP schemas supported are
+
+1. groups as first-class LDAP entries that contain lists of members of the group
+2. users as first-class LDAP entries that contain lists of groups they are a member of
+3. schema 2 above, also including first-class groups to store metainformation
+
+### Data Flow
+* Determine list of LDAP groups to be synced
+* Find LDAP group members
+* Determine OpenShift `Identities` from LDAP group members
+* Determine OpenShift `Users` from OpenShift `Identities`
+* Determine OpenShift `Group` from LDAP group
+* Populate new OpenShift `Group`'s `Users` from resulting OpenShift `UserIdentityMappings`, update OpenShift `Group` metadata fields tied to LDAP attributes, leave other OpenShift `Group` fields unchanged
+
+##  Determining Which LDAP Groups To Sync
+An `LDAPGroupLister` determines what LDAP groups needs to be synced and outputs the result as a set of unique identifier strings (for instance, strings that encode the value of the unique filter key needed to inject into the unique filter that, when conjoined with the common filter, will retrieve those groups from the LDAP server.) The other objects that take these unique identifiers need to understand the format of this string (e.g. these objects will be tightly coupled).
+
+```go
+// LDAPGroupLister lists the specific group filters for all groups that should be synced
+type LDAPGroupLister interface {
+  ListGroups() (groupUIDs []string], err error)
+}
+```
+
```

Comment:
enumerate the different impls you see:
 1. openshift group whitelist
 1. all openshift groups
 1. ldap group whitelist
 1. ldap groups minus (potentially empty) blacklist